skimage/filters/rank/generic.py

```diff
@@ -92,39 +92,40 @@ def _handle_input(image, selem, out, mask, out_dtype=None, pixel_size=1):
     is_8bit = image.dtype in (np.uint8, np.int8)
 
     if is_8bit:
-        max_bin = 255
+        n_bins = 256
     else:
-        max_bin = max(4, image.max())
+        # do the addition after the conversion to python to ensure
```

Comment:
I don't understand this comment.

Reply:
It is to avoid overflow when we add `1`
```
In [4]: np.empty(3, dtype='uint8').max().dtype                                
Out[4]: dtype('uint8')
```

Reply:
Clearly, i am wrong though!!!
In [6]: max(3, np.empty(3, dtype='uint8').max()).__class__                    
Out[6]: numpy.uint8